lib/pf/mfa/Akamai.pm

```diff
@@ -151,32 +152,43 @@ sub check_user {
             return $ACTIONS{'push'}->($self,$default_device[0]->{'device'},$username);
        }
     }
-    elsif ($self->radius_mfa_method eq 'strip-otp' || $self->radius_mfa_method eq 'second-password') {
+    elsif ($self->radius_mfa_method eq 'strip-otp' || $self->radius_mfa_method eq 'second-password' || $self->radius_mfa_method eq 'sms' || $self->radius_mfa_method eq 'phone') {
         if (defined $otp) {
             if ($otp =~ /^\d{6,6}$/ || $otp =~ /^\d{16,16}$/) {
                 if ( grep $_ eq 'totp', @{$default_device[0]->{'methods'}}) {
-                    return $ACTIONS{'totp'}->($self,$default_device[0]->{'device'},$username,$otp);
+                    return $ACTIONS{'totp'}->($self,$default_device[0]->{'device'},$username,$otp,$devices);
                 } else {
-                    $logger->warn("Unsuported method totp on device ".$default_device[0]->{'name'});
+                    $logger->info("Unsuported method totp on device ".$default_device[0]->{'name'});
                     return $FALSE;
                 }
             } elsif ($otp =~ /^\d{8,8}$/) {
-                    return $ACTIONS{'check_auth'}->($self,$default_device[0]->{'device'},$username,$otp);
+                $logger->info("OTP Verification");
+                return $ACTIONS{'check_auth'}->($self,$default_device[0]->{'device'},$username,$otp,$devices);
             } elsif ($otp =~ /^(sms|push|phone)(\d?)$/i) {
                 my @device = $self->select_phone($devices->{'result'}->{'devices'}, $2);
                 my $method = $1;
                 foreach my $device (@device) {
                     if ( grep $_ =~ $METHOD_ALIAS{$method}, @{$device->{'methods'}}) {
-                        return $ACTIONS{$method}->($self,$device->{'device'},$username,$1);
+                        return $ACTIONS{$method}->($self,$device->{'device'},$username,$1,$devices);
                     } else {
-                        $logger->warn("Unsuported method on device ".$device->{'name'});
+                        $logger->info("Unsuported method on device ".$device->{'name'});
                         return $FALSE;
                     }
                 }
             } else {
-                $logger->warn("Method not supported");
+                $logger->info("Method not supported");
                 return $FALSE;
             }
+        } elsif ($self->radius_mfa_method eq 'sms' || $self->radius_mfa_method eq 'phone') {
```

Comment:
This looks like dead code, as it will never be called because SMS and phone are included above.

Reply:
Ignore